static/js/settings.js

```diff
@@ -0,0 +1,118 @@
+const API_PREFIX = 'api'
+const CONTAINER_ID = 'settings'
+
+const baseUrl = window.location.origin
+const fields = ["session_id", "folder_path", "loop_delay_seconds", "loop_variation_percentage", "extra_ids"]
+
+const requestUrl = `${baseUrl}/${API_PREFIX}/settings/`
+const loggedinUrl = `${baseUrl}/${API_PREFIX}/loggedin/`
+
+
+const normalizeExtraIds = (extraIds) => (  // Splits a string of names/numbers into a list when it finds "," or "\n"
+    extraIds.split(/,|\n/).map(el => (el.trim()))
+)
+
+const checkLogin = async () => {
+    const isUserLoggedIn = await (await fetch(loggedinUrl)).json() // Asks the server if the user is logged in
+    const loginField = document.getElementById('login_field')
+    if (isUserLoggedIn["logged"]) loginField.style.display = 'none'  // If the user is not logged in shows the login form 
+    else loginField.style.display = 'block' 
+}
+
+const fetchResponseToHtml = async (response) => {  // Sets the placeholders and values in HTML
+    if (response.status !== 200) {
+        // TODO: Implement a proper error page
+        return
+    }
+
+    const responseData = await response.json()
+    checkLogin()
+
+    for (element in responseData) {  // For every element sets its value in HTML
+        const elementField = document.getElementById(element)
+        if (elementField) {
+            if (element === "extra_ids") {
+                elementField.value = responseData[element]
+            } else {
+                elementField.placeholder = responseData[element]
+            }
+        }
+    }
+}
+
+const renderClientSidePage = async () => {  // Makes the GET request and calls fetchResponseToHtml with response as arg
```

Comment:
renderSettingsPage? 

We should change it also for gallery, but let's start with this one!